### PR TITLE
Support `StructValue` in `StringifyTypeExpr`

### PR DESCRIPTION
**Pull request by josh11b**

### CODE REVIEW

File: toolchain/sem_ir/stringify_type.cpp [hunk 1]
```diff
@@ -441,6 +441,32 @@ auto StringifyTypeExpr(const SemIR::File& sem_ir, InstId outer_inst_id)
         }
         break;
       }
+      case CARBON_KIND(StructValue inst): {
+        auto field_values = sem_ir.inst_blocks().Get(inst.elements_id);
+        if (field_values.empty()) {
+          out << "{}";
+          break;
+        }
+        auto struct_type = sem_ir.types().GetAs<StructType>(inst.type_id);
```
Comment (zygoloid) [inline toolchain/sem_ir/stringify_type.cpp]: We build `StructValue`s of class type for class values too. We also build them for a few other types that "should be" class types -- for the values of `FunctionDecl`s, `ClassDecl`s, and `InterfaceDecl`s -- but those are all empty so don't get here.

I think we can handle the class type case with simply:
```suggestion
        auto struct_type = sem_ir.types().GetAs<StructType>(
            sem_ir.types().GetObjectRepr(inst.type_id));
```

**Review by zygoloid — APPROVED** [review]
Looks good. Can you add a test for the class value printing case somewhere too?

**Comment by josh11b** [comment]
> Looks good. Can you add a test for the class value printing case somewhere too?

Not sure how to. If I try to create a type with a class-valued parameter, I get an error `argument for generic parameter is not a compile-time constant`.

**Comment by zygoloid** [comment]
Hm. I can't find a way either. We do generate such constants, but only as the value of an initializing expression, and I can't find a way to get the toolchain to include one of those in a diagnostic.

**Pull request merged**